playwright-gh-pages/README.md

```diff
@@ -0,0 +1,223 @@
+# Publishing Playwright reports to Github Pages
+
+When testing a Grafana plugin using the [`@grafana/plugin-e2e`](https://www.npmjs.com/package/@grafana/plugin-e2e?activeTab=readme) package, it is highly recommended to run tests against a matrix of Grafana versions (as demonstrated in this [example](https://grafana.com/developers/plugin-tools/e2e-test-a-plugin/ci) in the documentation). Each test run in this matrix generates an HTML report. By uploading these reports to a static site hosting service, they become immediately accessible for direct browsing, eliminating the need to download and serve them locally. This enhances productivity and fosters collaborative troubleshooting by making the results easily shareable and reviewable.
+
+This set of GitHub Actions streamlines the process of managing Playwright test reports. It automates uploading reports as artifacts, publishing them to GitHub Pages, and providing links in pull request comments. These actions work seamlessly together, enhancing collaboration, traceability, and test result management.
+
+## Overview
+
+The workflow consists of two main actions:
+
+1. **Upload Report Artifacts Action**: This action uploads test reports and summaries as GitHub artifacts. It can be used together with the `deploy-report-pages` action to publish the reports to GitHub Pages.
+2. **Deploy to GitHub Pages Action**: This action publishes the test artifacts to GitHub Pages and comments on the pull request with the results and corresponding links. It also cleans up and deletes old reports based on the specified retention policy.
+
+## Features
+
+- **Upload Report Artifacts Action**:
+
+  - Uploads test reports and summaries as GitHub artifacts.
+  - Supports conditional uploading based on test outcomes.
+  - Structures reports in a well-organized directory format, ensuring uniqueness for each test setup.
+
+- **Deploy to GitHub Pages Action**:
+  - Downloads test artifacts and publishes them to GitHub Pages.
+  - Comments on the pull request with the test results and links to the reports.
+  - Supports retention of reports for a specified number of days.
+
+## GitHub Pages Branch
```

Comment:
probably worth mentioning that github pages only work from one branch as a source. so if someone is already using github pages for something they need to either figure out how to combine their content with e2e reports or replace the content with e2e reports.

Reply:
Done https://github.com/grafana/plugin-actions/pull/50/commits/ec6df307c3ace7c4286f12cf6f414d316d744439